app/scripts/modules/netflix/fastProperties/domain/scope.domain.ts

```diff
@@ -0,0 +1,71 @@
+import {IPlatformProperty} from './platformProperty.model';
+import {CATEGORY} from '../scope/fastPropertyScopeSearchCategory.service';
+import {IImpactCount} from './impactCounts.interface';
+
+export class Scope {
+  public env: string;
+  public region: string;
+  public appId: string;
+  public appIdList: string[] = [];
+  public stack: string;
+  public cluster: string;
+  public asg: string;
+  public serverId: string;
+  public zone: string;
+  public instanceCounts: IImpactCount = <IImpactCount>{
+    down: 0,
+    up: 0,
+    failed: 0,
+    outOfService: 0,
+    starting: 0,
+    succeeded: 0,
+    total: 0,
+    unknown: 0
+  };
+
+  static build(platformProperty: IPlatformProperty) {
+    let scope = new Scope();
+    scope.env = platformProperty.env;
+    scope.region = platformProperty.region;
+    scope.appId = platformProperty.appId;
+    scope.stack = platformProperty.stack;
+    scope.asg = platformProperty.asg;
+    scope.serverId = platformProperty.serverId;
+    scope.zone = platformProperty.zone;
+    scope.cluster = platformProperty.cluster;
+
+    return scope;
+  }
+
+  public getBaseOfScope() {
```

Comment:
this "feels" wrong but I don't have a good answer on how to make this different.  a priority queue comes to mind, but seems like that would add more overhead...